src/lib/nlcManager.js

```diff
@@ -142,47 +194,38 @@ NLCManager.prototype._startTraining = function(){
 		if (this.classifierTraining) {
 			resolve(this.classifierTraining);
 		}
+		// Training with data initialized in opts.
 		else if (this.opts.training_data) {
 			let params = {
 				language: this.opts.classifierLanguage,
 				name: this.opts.classifierName,
 				training_data: this.opts.training_data
 			};
 
-			this.nlc.create(params, (err, response) => {
-				if (err) {
-					reject('Error creating classifier from provided training_data:' + JSON.stringify(err, null, 2));
-				}
-				else {
-					this.classifierTraining = response;
-					resolve(response);
-				}
+			this._createClassifier(params).then((result) => {
+				resolve(result);
 			});
 		}
+
+		// Training data from PouchDB.
 		else {
 			return nlcDb.open().then((db) => {
-				return db.getClasses();
-			})
-			.then((csvInput) => {
-				stringify(csvInput, (err, csvStream) => {
-					if (err){
-						reject('Error generating training data in csv format.');
-					}
+				return db.getClasses().then((csvInput) => {
+					stringify(csvInput, (err, csvStream) => {
+						if (err){
+							logger.error(`${TAG}: Error generating training data in csv format.`, err);
+							reject('Error generating training data in csv format.');
+						}
 
-					let params = {
-						language: this.opts.classifierLanguage,
-						name: this.opts.classifierName,
-						training_data: csvStream
-					};
+						let params = {
+							language: this.opts.classifierLanguage,
+							name: this.opts.classifierName,
+							training_data: csvStream
+						};
 
-					this.nlc.create(params, (err, response) => {
-						if (err) {
-							reject('Error creating classifier from nlcDb:' + JSON.stringify(err, null, 2));
-						}
-						else {
-							this.classifierTraining = response;
-							resolve(response);
-						}
+						this._createClassifier(params).then((result) => {
+							resolve(result);
+						});
```

Comment:
Also need a catch clause here.  The failure of inner promise will not propagate to the outer promise chain.